rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/cu_components.kt

```diff
@@ -42,6 +42,7 @@ data class CompilationUnit(
     // necessary to avoid that during data references resolving, are considered expression declared within procedures as well.
     @property:Link val procedures: List<CompilationUnit>? = null,
     val procedureName: String? = null,
+    val isFakeProcedure: Boolean = false,
```

Comment:
I am not sure if this new property is really necessary, but maybe I am wrong.
If you aim to determine whether 'CompilationUnit' is a procedure or not, I think you may test 'CompilationUnit.procedureName', which is not null only if the 'CompilationUnit' is related to a procedure.

Reply:
Is necessary because there isn't any flag that indicates if a procedure is "real" or "fake". With a flag like that, I can resolve a procedure declared like fake but with its implementation imported after and by directive.
I can try to make that attribute optional, with a value if the `CompilationUnit` refers to a procedure.